src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -214,14 +215,26 @@ public ValueCollection Values
         {
             get
             {
+#if DEBUG
+                int version = _version;
```

Comment:
if you're using the existing `_version` field doesn't it have to be volatile or incremented with Interlocked?

@stephentoub

Reply:
In `DEBUG` its incremented in `ConcurrentAccessWriteCheck` with
```csharp
Interlocked.CompareExchange(ref _version, version + 1, version) != version
```

Reply:
Changed to
```csharp
if (Interlocked.Exchange(ref _version, version + 1) != version)
{
    ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported();
}
```
As don't care if it stops version being updated; just that it throws if the version wasn't correct